tracer/src/Datadog.Trace/Logging/Internal/TextFormatters/SingleLineTextFormatter.cs

```diff
@@ -0,0 +1,50 @@
+﻿// <copyright file="SingleLineTextFormatter.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System.IO;
+using Datadog.Trace.Util;
+using Datadog.Trace.Vendors.Serilog.Events;
+using Datadog.Trace.Vendors.Serilog.Formatting;
+using Datadog.Trace.Vendors.Serilog.Formatting.Display;
+
+namespace Datadog.Trace.Logging.Internal.TextFormatters;
+
+internal class SingleLineTextFormatter : ITextFormatter
+{
+    public void Format(LogEvent logEvent, TextWriter output)
+    {
+        if (logEvent == null!) { ThrowHelper.ThrowArgumentNullException(nameof(logEvent)); }
+        if (output == null!) { ThrowHelper.ThrowArgumentNullException(nameof(output)); }
+
+        // buffer the entire log event into a single line before writing it to the output
+        var buffer = StringBuilderCache.Acquire();
+        var bufferWriter = new StringWriter(buffer, output.FormatProvider);
+
+        var utcTimestamp = logEvent.Timestamp.ToUniversalTime();
+        var logLevel = LevelOutputFormat.GetLevelMoniker(logEvent.Level, "u3");
+        bufferWriter.Write($"[{utcTimestamp:yyyy-MM-dd HH:mm:ss.fff zzz} | DD_TRACE_DOTNET | {logLevel}] ");
+
+        // write the message to the output, using the template and properties
+        logEvent.RenderMessage(bufferWriter, output.FormatProvider);
+
+        // if there is an exception, write it to the output
+        if (logEvent.Exception != null)
+        {
+            bufferWriter.Write($" | {logEvent.Exception}");
+        }
+
+        // replace any newlines with literal "\n" to ensure the output is a single line
+        buffer.Replace("\r\n", "\\n")
+              .Replace("\n", "\\n");
```

Comment:
eww 😅 That should make it nice and hard to read 😛

Reply:
>= 6.0 we can use: https://learn.microsoft.com/en-us/dotnet/api/system.string.replacelineendings?view=net-6.0

Reply:
`ReplaceLineEndings()` allocates a new string. I was doing these `Replace()` calls on a `StringBuilder`. Does we care?

Reply:
Switched to `ReplaceLineEndings()` in .NET 6+, thanks.

Reply:
Ah, yeah, you're right 👍  but you know what would clear that up for sure, a unit test 😉

Reply:
added a unit test